scripts/kmodtool

```diff
@@ -116,7 +116,7 @@ Provides:        kmod-${kmodname}-PAE = %{?epoch:%{epoch}:}%{version}-%{release}
 Requires:        akmod-${kmodname} = %{?epoch:%{epoch}:}%{version}-%{release}
 EOF
 
-	if [[ ${obsolete_name} ]]; then
+	if [[ -n ${obsolete_name} ]]; then
```

Comment:
`[ -n "$obsolete_name" ]`

Reply:
Newer versions of ShellCheck (0.8+) tend to advise for `[[` usage in Bash scripts https://github.com/koalaman/shellcheck/wiki/SC2292

Variable interpolation tend to be done with curly braces across the script
```
# grep -c '\$[{]' kmodtool 
198
# grep -c '\$[^{]' kmodtool 
15
```

Reply:
The `$`/`${` distinction is fair enough here, yeah. OTOH, as far as I can tell, you're already touching almost-all `[[` specials anyway, and a cursory glance at the shellcheck output shows that these are the vast majority of the non-/bin/shisms (with a few `local`s, a lot of `==`s, and a `$'\n'`) – it may be worth-while to mass-replace `[[`s with `[`s and upgrade the shebang to `/bin/sh`?

Reply:
While `[[`, `==` and `$''` are mechanical changes, there's also `shopt` for extended globs. Looks like this functionality is paramount for the tool to work properly. Refactoring this functionality is way out of the scope of this PR.
I'd upload the mechanical changes regardless, it would leave us with `local` and `shopt` to remove.

https://github.com/openzfs/zfs/blob/master/scripts/kmodtool#L569-L575